internal/controller/postgrescluster/cluster.go

```diff
@@ -288,6 +289,7 @@ func (r *Reconciler) reconcileDataSource(ctx context.Context,
 	// in place (and therefore recreating the data directory).  If the user hasn't requested
 	// PG data initialization or an in-place restore, then simply return.
 	var dataSource *v1beta1.PostgresClusterDataSource
+	var cloudDataSource *v1beta1.PGBackRestDataSource
```

Comment:
Except for the fact that it might be easier for users if we have PostgresCluster and PGBackRest as two separate fields in DataSource, I am coming around to the idea that we should create one generic object that can accommodate both cases. Is this worth pursuing at this time however? And how confusing might that be to our end-users?

Reply:
My $0.02, I think the generic object may well be a good idea, but I think it's something we'd want to plan and tackle at a later point (if at all). Definitely something we should discuss further.

Reply:
I put together a clean version with a generic for both PostgresCluster restores and pgBackRest restores, and it felt more cumbersome in some ways. (The conversation on Friday helped to clarify this a bit in terms of distinguishing whether you're pointing to a cluster or to a pgBackRest backup, which made me feel better about leaving them separate for now, but definitely think we might want to revisit this topic.)